src/Microsoft.ML.Data/Transforms/TermTransform.cs

```diff
@@ -175,32 +192,180 @@ private static VersionInfo GetTermManagerVersionInfo()
                 loaderSignature: TermManagerLoaderSignature);
         }
 
-        // These are parallel to Infos.
-        private readonly ColumnType[] _types;
-        private readonly BoundTermMap[] _termMap;
+        private readonly TermMap[] _unboundMaps;
         private readonly bool[] _textMetadata;
-
         private const string RegistrationName = "Term";
 
-        private static volatile MemoryStreamPool _codecFactoryPool;
-        private volatile CodecFactory _codecFactory;
+        private static (string input, string output)[] GetColumnPairs(ColumnInfo[] columns)
+        {
+            Contracts.CheckValue(columns, nameof(columns));
+            return columns.Select(x => (x.Input, x.Output)).ToArray();
+        }
+
+        private readonly ColumnInfo[] _columns;
+
+        public IReadOnlyCollection<ColumnInfo> Columns => _columns.AsReadOnly();
 
-        private CodecFactory CodecFactory
+        private ColInfo[] CreateInfos(ISchema schema)
         {
-            get
+            Host.AssertValue(schema);
+            var infos = new ColInfo[ColumnPairs.Length];
+            for (int i = 0; i < ColumnPairs.Length; i++)
             {
-                if (_codecFactory == null)
+                if (!schema.TryGetColumnIndex(ColumnPairs[i].input, out int colSrc))
+                    throw Host.ExceptUserArg(nameof(ColumnPairs), "Source column '{0}' not found", ColumnPairs[i].input);
+                var type = schema.GetColumnType(colSrc);
+                string reason = TestIsKnownDataKind(type);
+                if (reason != null)
+                    throw Host.ExceptUserArg(nameof(ColumnPairs), InvalidTypeErrorFormat, ColumnPairs[i].input, type, reason);
+                infos[i] = new ColInfo(ColumnPairs[i].output, ColumnPairs[i].input, type);
+            }
+            return infos;
+        }
+
+        public TermTransform(IHostEnvironment env, IDataView input,
+            ColumnInfo[] columns,
+            string file = null, string termsColumn = null,
+            IComponentFactory<IMultiStreamSource, IDataLoader> loaderFactory = null)
```

Comment:
let's remove these params

Reply:
I still need them, but I can make this private

---
In reply to: [214387120](https://github.com/dotnet/machinelearning/pull/759#discussion_r214387120) [](ancestors = 214387120)

Reply:
who is using them now?

---
In reply to: [214422979](https://github.com/dotnet/machinelearning/pull/759#discussion_r214422979) [](ancestors = 214422979,214387120)

Reply:
Anyone who use arguments and command line. 
Codebase wise - WordHashBag/WordBag still use Arguments class with DataFile

---
In reply to: [214479628](https://github.com/dotnet/machinelearning/pull/759#discussion_r214479628) [](ancestors = 214479628,214422979,214387120)